### DISCUSSION TITLE
Preparing wheel metaadata error while installing on restricted systems

### DISCUSSION THREAD

**Discussion by photonfly** [cocotb/cocotb#2805] [Q&A] [answered]
Hi, I download the version 1.5.2 code, unzip the code in the restriced Red hat system , and install using: "pip install --no-build-isolation".
It shows:
![image](https://user-images.githubusercontent.com/22272733/145349608-774612de-d514-4560-95d5-896b0144e3d0.png)
I also tried the instructions at: https://github.com/cocotb/cocotb/wiki/Tier-2-Setup-Instructions
But I did not find the file name: "_version.py". Looking for your reply.

**Comment by imphil** [comment, marked as answer]
version.py is created on demand in 1.5.2 (by setuptools_scm) and requires "pip install" to be called from a Git source repository. The probably easiest option for you is to use the 1.6.x release, where the source tree contains this file.

**Reply by photonfly** [reply]
Thanks! However, while using the latest version: 1.6.0, same error happens:
![image](https://user-images.githubusercontent.com/22272733/145366796-56a53480-f2cf-487f-9833-bc7be8b79e8b.png), how can I trouble shoot it?

**Reply by imphil** [reply]
It looks like you're missing the wheels package, which provides `bdist_wheel`.

The wheel package can either be installed through your distributions' package manager (yum or dnf in your case most likely), through pip (which probably doesn't work in your environment?), or maybe your IT department has already made it available somewhere.

Here are some options, you need to find the one that works for you :-)
```
# Through pip
pip install wheel

# Through yum (e.g. for RHEL 8 and probably some earlier versions)
yum install python3-wheel
```

**Reply by ktbarrett** [reply]
You might be able to also install cocotb without first installing the wheel package by running the setup.py directly using `python setup.py install` when at the root of the source directory.

**Reply by photonfly** [reply]
Thank for your replying ktbarrett!
After updating my wheel package to version 0.37.0, I tried installing using pip and Python3 setup.py install, they both lead to error: "**ERROR: Failed building wheel for cocotb**",   
![image](https://user-images.githubusercontent.com/22272733/145556211-bd9b763e-61c9-41e8-9ec0-0a21acd918e1.png)
It also reports **cannot find -lpython3.6m**
![image](https://user-images.githubusercontent.com/22272733/145556270-a064f80c-a099-4f2d-9a33-85c3b15480e1.png)

Is it still missing any package?

**Reply by ktbarrett** [reply]
Please see the [installation instructions](https://docs.cocotb.org/en/stable/install.html). You need the Python development package.
